x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/result_settings_table/result_settings_non_text_fields_body.tsx

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+
+import { useValues, useActions } from 'kea';
+
+import {
+  EuiTableBody,
+  EuiTableRow,
+  EuiTableRowCell,
+  EuiCheckbox,
+  EuiTableRowCellCheckbox,
+} from '@elastic/eui';
+
+import { ResultSettingsLogic } from '..';
+import { FieldResultSetting } from '../types';
+
+export const ResultSettingsNonTextFieldsBody: React.FC = () => {
+  const { nonTextResultFields } = useValues(ResultSettingsLogic);
+  const { toggleRawForField } = useActions(ResultSettingsLogic);
+
+  const resultSettingsArray: Array<[string, Partial<FieldResultSetting>]> = useMemo(() => {
+    return Object.entries(nonTextResultFields).sort(([aFieldName], [bFieldName]) =>
+      aFieldName > bFieldName ? 1 : -1
+    );
+  }, [nonTextResultFields]);
+
+  return (
+    <EuiTableBody>
+      {resultSettingsArray.map(([fieldName, fieldSettings]) => (
+        <EuiTableRow key={fieldName}>
+          <EuiTableRowCell data-test-subj="ResultSettingFieldName">{fieldName}</EuiTableRowCell>
```

Comment:
Same as above, minor UI suggestion

```suggestion
          <EuiTableRowCell data-test-subj="ResultSettingFieldName">
            <code>{fieldName}</code>
          </EuiTableRowCell>
```